In [1]:
import pandas as pd
import ast
from data import data
pd.options.display.max_rows = 500

In [2]:
books = data.loadAndClean('../booksummaries/booksummaries.txt')

After loading and cleaning the data set I want to do some analysis on the book genres and their combinations

In [3]:
books = books[['bookGenre', 'plotSum']]

In [4]:
books.shape

(12841, 2)

We have 12841 books and their summaries after dropping the NA's (done in the loadAndClean function).  
Let's key in on the genres now.

In [5]:
# genres is a pandas.DataFrame which is a list of every single genre\
# mentioned in the initial booksummaries DataFrame
genres = pd.DataFrame( v for v in books.bookGenre.sum())
genres.columns = ['bookGenre']

In [6]:
genres.groupby('bookGenre')['bookGenre'].count().to_frame().sort_values('bookGenre', ascending=False).head(5)

/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: 'bookGenre' is both an index level and a column label.
Defaulting to column, but this will raise an ambiguity error in a future version
  """Entry point for launching an IPython kernel.


bookGenre
bookGenre                     
fiction                   4747
speculative_fiction       4314
science_fiction           2870
novel                     2463
fantasy                   2413

In [7]:
genres.groupby('bookGenre')['bookGenre'].count().shape

(227,)

These above two cells show us the top 5 genres by raw count, and also that we have 227 unique genres we are dealing with and that we need to roll up.